content/docs/faq-functions.md

```diff
@@ -177,3 +177,69 @@ class Alphabet extends React.Component {
   }
 }
 ```
+
+### How can I prevent a function from being called too quickly or too many times in a row?
+
+If you have an event handler such as `onClick` or `onScroll` and want to prevent the callback from being fired too quickly, you can wrap the handler with a utility such as [`_.debounce`](https://lodash.com/docs#debounce) or [`_.throttle`](https://lodash.com/docs#throttle). See [this visualization](http://demo.nimius.net/debounce_throttle/) for a comparison of the two.
+
+> Note:
+>
+> Both `_.debounce` and `_.throttle` provide a `cancel` method to cancel delayed callbacks. You should either call this method from `componentWillUnmount` _or_ check to ensure that the component is still mounted within the delayed function.
+
+#### Throttle
+
+Throttling prevents a function from being called more than once in a given window of time. The example below throttles a "click" handler to prevent calling it more than once per second.
+
+```jsx
+import throttle from "lodash.throttle";
+
+class LoadMoreButton extends React.Component {
+  componentWillUnmount() {
+    this._handleClick.cancel();
+  }
+
+  render() {
+    return <button onClick={this._handleClick}>Load More</button>;
+  }
+
+  _handleClick = throttle(() => {
```

Comment:
This uses a syntax that isn't part of any ES standard. So far we've avoided such syntax anywhere else in the documentation except the page specifically explaining it (with an "Experimental" warning).

Are we making an exception here? Or do we intentionally want to start using unstable syntax elsewhere in the docs (e.g. on the homepage)? How do we decide when to use it?

Reply:
I'll send a PR since it's not essential here.

Reply:
Good question, Dan.

This example uses the class public fields proposal (currently in Stage 2) because of its clarity.

I was under the impression we used this syntax elsewhere (like on the ["React Without ES6"](https://reactjs.org/docs/react-without-es6.html) and ["Component State"](https://reactjs.org/docs/faq-state.html) pages) but it's not very common.

Reply:
> I'll send a PR since it's not essential here.

Sounds good to me 👍

Reply:
On https://reactjs.org/docs/react-without-es6.html, it specifically calls out:

```js
  // WARNING: this syntax is experimental!
  // Using an arrow here binds the method:
```

and mentions:

>you may enable the **experimental** Class Properties syntax proposal with Babel:

https://reactjs.org/docs/faq-state.html is also a new page so it isn't consistent with the original docs either.

Reply:
There is an explanation for the class-fields-for-bound-functions on this same FAQ page, so it seems like fair game. But if it clears things up, sure, bind in constructor would be fine.

Reply:
Yes, I saw the note on the ES6 page. And as I said above, I'm fine with removing the example.

Reply:
As a side note, what do you think of _defining_ bound functions in the constructor instead of binding the prototype methods?

```js
class Comp extends React.Component {
  constructor(props) {
    super(props)
    this.handleFoo = () => { /* method definition, already bound */ }
  }
}
```

Reply:
I'm all for mentioning it :-) Perhaps we could have a "how to write more concise code" page where we explain all those tricks. But IMO code from doc pages need to be copy-pasteable into browsers, fiddles, editors, etc, without knowledge of special proposals.